legate/core/legion.py

```diff
@@ -1017,14 +1017,42 @@ def __init__(self, context, runtime, owned=True):
             The Legion context from get_legion_context()
         runtime : legion_runtime_t
             Handle for the Legion runtime from get_legion_runtime()
+        handle : legion_field_space_t
+            Created handle for a field space from a Legion C API call
+        owned : bool
+            Whether this object owns the handle for this field space and
+            can delete the field space when the object is collected
         """
         self.context = context
         self.runtime = runtime
-        self.handle = legion.legion_field_space_create(runtime, context)
+        self.fields = dict()
+        if handle is None:
+            handle = legion.legion_field_space_create(runtime, context)
+        else:
+            num_fields = legion.legion_field_space_get_fields_size(
+                runtime,
+                context,
+                handle,
+            )
+            fids = ffi.new("legion_field_id_t[%d]" % num_fields)
+            legion.legion_field_space_get_fields(
+                runtime,
+                context,
+                handle,
+                fids,
+                num_fields,
+            )
+            for fid in fids:
+                self.fields[fid] = legion.legion_field_id_get_size(
+                    runtime,
+                    context,
+                    handle,
+                    fid,
+                )
```

Comment:
I feel like this code is actually unnecessary and potentially unsafe. It's not clear that we can actually add/use these fields to our data structure since we're not sure who or what made the fields. For certain we can't delete these fields at any time since we didn't make them. I would be in favor of just ignoring any fields already in the field space. We could also provide a way for the client to pass in an optional list of fields that are already in in the field space that we are going to take ownership for.

Reply:
The user whose request motivated this change (see https://github.com/nv-legate/legate.core/issues/12) only needs to initialize a `FieldSpace` object from a `legion_field_space_t` so he can use that on a `Region` constructor, and won't be querying the fields on this `FieldSpace`, or creating new fields. Therefore it was sufficient to simply disallow such operations on `FieldSpace`s created from a handle.

@lightsighter Take a look if you can and let me know if the changes make sense to you.